srml/contracts/src/rent.rs

```diff
@@ -137,7 +137,11 @@ fn try_evict_or_and_pay_rent<T: Trait>(
 			dues_limited < rent_budget < balance - subsistence < balance - existential_deposit;
 			qed",
 		);
-	}
+
```

Comment:
why don't you just handle the imbalance here ? with `T::RentPayment::on_unbalanced(imbalance);` directly.

Reply:
That works!

Reply:
The tests seem to fail due to the withdrawn amount being zero: 
```
---- tests::restorations_dirty_storage_and_different_storage stdout ----
thread 'tests::restorations_dirty_storage_and_different_storage' panicked at 'assertion failed: `(left != right)`
  left: `0`,
 right: `0`', srml/contracts/src/tests.rs:305:3
```

Reply:
does in case of insufficient_rent the imbalance be dropped instead of put in RentPayment ?

Reply:
but indeed the withdraw is correctly 0 as the rent_budget of the contract is zero.

I'm not sure if the test is relevant, if yes then we can just disable non null withdraw in case of this restoration tests.
you can do it with:
```diff
@@ -297,12 +297,16 @@ fn compile_module<T>(wabt_module: &str)
        Ok((wasm, code_hash))
 }
 
+pub const DISABLE_NON_ZERO_RENT_ASSERT: AtomicBool = AtomicBool::new(false);
+
 // A hook for rent payments to check that the rent is not zero.
 pub struct NonZeroRentHook;
 
 impl OnUnbalanced<NegativeImbalanceOf<Test>> for NonZeroRentHook {
        fn on_unbalanced(amount: NegativeImbalanceOf<Test>) {
-               assert_ne!(amount.peek(), BalanceOf::<Test>::zero());
+               if !DISABLE_NON_ZERO_RENT_ASSERT.load(..) {
+                       assert_ne!(amount.peek(), BalanceOf::<Test>::zero());
+               }
        }
 }
```

Reply:
Ok, good idea.

Reply:
I think that for the time being, I'm just going to get rid of the test though.